trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/model/ConfigurationQueryParameters.java

```diff
@@ -23,9 +25,36 @@
 public interface ConfigurationQueryParameters {
 
     /**
-     * @return The query parameter map.
+     * @return The parameters.
      */
     @NonNull
-    @Schema(required = true, description = "Parameters as specified in corresponding ConfigurationTypeDescriptor. Use key `path` for file URI string values.")
-    Map<String, Object> getParameters();
+    @Schema(required = true)
+    ConfigurationParameters getParameters();
+
+    /**
+     * No expected properties below, as per current trace-server protocol.
+     */
+    interface ConfigurationParameters {
+        /**
+         * @return the name of the configuration
+         */
+        @Schema(required = true, description = "Unique name of the configuration. If omitted a unique name will be generated.")
+        String getName();
+        /**
+         * @return the description of the configuration
+         */
+        @Schema(required = false, description = "Optional description of the configuration.")
+        String getDescription();
+        /**
+         * @return the typeId of the configuration according to the {@link ConfigurationSourceType}
+         */
+        @Schema(required = false, description = "Optional typeId of the configuration according to the corresponding ConfigurationTypeDescriptor. Omit if it's part of the endpoint URI.")
+        @JsonProperty("typeId")
+        String getTypeId();
+        /**
+         * @return properties map as defined in the corresponding {@link ConfigurationSourceType}
+         */
+        @Schema(required = true, description = "Properties as specified in the schema or list of ConfigurationParameterDescriptor of the corresponding ConfigurationTypeDescriptor.")
```

Comment:
Thinking about it a bit more we should call this `parameters` instead of `properties`, because the resulting Configuration has a member called `parameters` instead of properties. It would be not clear otherwise.

Reply:
Done

Reply:
I'm OK with that change.